server/src/test/java/org/elasticsearch/script/ScriptMetaDataTests.java

```diff
@@ -168,6 +171,41 @@ public void testLoadEmptyScripts() throws IOException {
         assertWarnings("empty templates should no longer be used");
     }
 
+    public void testOldStyleDropped() throws IOException {
+        XContentBuilder builder = XContentBuilder.builder(XContentType.JSON.xContent());
+
+        builder.startObject();
+        {
+            builder.startObject("painless#test");
+            {
+                builder.field("lang", "painless");
+                builder.field("source", "code");
+            }
+            builder.endObject();
+            builder.startObject("lang#test");
+            {
+                builder.field("lang", "test");
+                builder.field("source", "code");
+            }
+            builder.endObject();
+            builder.startObject("test");
+            {
+                builder.field("lang", "painless");
+                builder.field("source", "code");
+            }
+            builder.endObject();
+        }
+        builder.endObject();
+
+        XContentParser parser = XContentType.JSON.xContent()
+                .createParser(NamedXContentRegistry.EMPTY, DeprecationHandler.THROW_UNSUPPORTED_OPERATION,
+                        BytesReference.bytes(builder).streamInput());
+        ScriptMetaData smd = ScriptMetaData.fromXContent(parser);
+        assertNull(smd.getStoredScript("painless#test"));
+        assertNull(smd.getStoredScript("lang#test"));
```

Comment:
Maybe check the size of the Map returned from `getStoredScripts()`?

Reply:
Good call. Added.